src/bin/ion/commands/beta/generate/generator.rs

```diff
@@ -213,245 +332,20 @@ impl<'a> CodeGenerator<'a> {
             }
         } else {
-            code_gen_context.with_data_model(current_data_model);
+            code_gen_context.with_abstract_data_type(current_abstract_data_type);
         }
         Ok(())
     }
-
-    /// Generates a struct field based on field name and value(data type)
-    fn generate_struct_field(
-        &mut self,
-        tera_fields: &mut Vec<Field>,
-        isl_type_ref: &IslTypeRef,
-        modules: &mut Vec<String>,
-        field_name: &str,
-        imports: &mut Vec<Import>,
-        code_gen_context: &mut CodeGenContext,
-    ) -> CodeGenResult<()> {
-        let value = self.generate_field_value(isl_type_ref, modules, imports, code_gen_context)?;
-
-        tera_fields.push(Field {
-            name: {
-                match self.language {
-                    Language::Rust => field_name.to_case(Case::Snake),
-                    Language::Java => field_name.to_case(Case::Camel),
-                }
-            },
-            value,
-        });
-        Ok(())
-    }
-
-    /// Generates field value in a struct which represents a data type in codegen's programming language
-    fn generate_field_value(
-        &mut self,
-        isl_type_ref: &IslTypeRef,
-        modules: &mut Vec<String>,
-        imports: &mut Vec<Import>,
-        code_gen_context: &mut CodeGenContext,
-    ) -> CodeGenResult<String> {
-        Ok(match isl_type_ref {
-            IslTypeRef::Named(name, _) => {
-                if !self.is_built_in_type(name) {
-                    imports.push(Import {
-                        module_name: name.to_case(Case::Snake),
-                        type_name: name.to_case(Case::UpperCamel),
-                    });
-                }
-                let schema_type: IonSchemaType = name.into();
-                self.generate_sequence_field_value(
-                    schema_type.target_type(&self.language).to_string(),
-                    code_gen_context,
-                )
-            }
-            IslTypeRef::TypeImport(_, _) => {
-                unimplemented!("Imports in schema are not supported yet!");
-            }
-            IslTypeRef::Anonymous(type_def, _) => {
-                self.anonymous_type_counter += 1;
-                self.generate_data_model(modules, type_def)?;
-                let name = format!("AnonymousType{}", self.anonymous_type_counter);
-                imports.push(Import {
-                    module_name: name.to_case(Case::Snake),
-                    type_name: name.to_case(Case::UpperCamel),
-                });
-                self.generate_sequence_field_value(name, code_gen_context)
-            }
-        })
-    }
-
-    /// Generates an appropriately typed sequence in the target programming language to use as a field value
-    pub fn generate_sequence_field_value(
```

Comment:
(🗺️ PR tour)`generate_sequence_field_name` has been moved to new language based implementation `Language::target_type_as_sequence` which returns sequence type based on programming language. 

e.g.
     target_type = "Foo" returns "ArrayList<Foo>"
     target_type = "Foo" returns "Vec<Foo>"

Reply:
~~RE: `Language::target_type_as_sequence`: is it true that for any language `L`, no matter what `target_type` is the return value will be the same?~~

~~Said another way, could we simplify this method to `Language::sequence_type`?~~

Having now read down to the implementation, I see that it's making a sequence with `target_type` as the type parameter. (`ArrayList<Foo>`, `Vec<Foo>`).

How about `sequence_of_target_type`?